src/libs/actions/Report.ts

```diff
@@ -1607,6 +1609,66 @@ function readNewestAction(reportID: string | undefined, shouldResetUnreadMarker
     }
 }
 
+function markAllMessagesAsRead() {
+    if (isAnonymousUser()) {
+        return;
+    }
+
+    const newLastReadTime = DateUtils.getDBTimeWithSkew();
+
+    type PartialReport = {
+        lastReadTime: Report['lastReadTime'] | null;
+    };
+    const optimisticReports: Record<string, PartialReport> = {};
+    const failureReports: Record<string, PartialReport> = {};
+    const reportIDList: string[] = [];
+    Object.values(allReports ?? {}).forEach((report) => {
+        if (!report) {
+            return;
+        }
+
+        const oneTransactionThreadReportID = ReportActionsUtils.getOneTransactionThreadReportID(
```

Comment:
@bernhardoj Is there a reason for doing this? Is there a problem if we mark both report and expense as read in one-transaction view?

Reply:
@bernhardoj Can you mark the code where I can identify this case?